src/backup.ts

```diff
@@ -0,0 +1,520 @@
+  /**
+   * @typedef {array} GetMetadataResponse
+   * @property {object} 0 The {@link Backup} metadata.
+   * @property {object} 1 The full API response.
+   */
+  /**
+   * @callback GetMetadataCallback
+   * @param {?Error} err Request error, if any.
+   * @param {object} metadata The {@link Backup} metadata.
+   * @param {object} apiResponse The full API response.
+   */
+  /**
+   * Retrieves backup's metadata.
+   *
+   * @see {@link #getState}
+   * @see {@link #getExpireTime}
+   *
+   * @method Backup#getMetadata
+   * @param {object} [options] Configuration object.
+   * @param {GetMetadataCallback} [callback] Callback function.
+   * @returns {Promise<GetMetadataResponse>}
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backup = instance.backup('my-backup');
+   * const [backupInfo] = await backup.getMetadata();
+   * console.log(`${backupInfo.name}: size=${backupInfo.sizeBytes}`);
+   */
+  getMetadata(
+    optionsOrCallback?: CallOptions | GetMetadataCallback,
+    cb?: GetMetadataCallback
+  ): void | Promise<GetMetadataResponse> {
+    const callback =
+      typeof optionsOrCallback === 'function'
+        ? (optionsOrCallback as GetMetadataCallback)
+        : cb;
+    const gaxOpts =
+      typeof optionsOrCallback === 'object'
+        ? (optionsOrCallback as CallOptions)
+        : {};
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IGetBackupRequest = {
+      name: this.formattedName_,
+    };
+    return this.request<IBackupTranslatedEnum>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'getBackup',
+        reqOpts,
+        gaxOpts,
+      },
+      (err, response) => {
+        callback!(err, response);
+      }
+    );
+  }
+
+  /**
+   * Retrieves the state of the backup.
+   *
+   * The backup state indicates if the backup has completed.
+   *
+   * @see {@link #getMetadata}
+   *
+   * @method Backup#getState
+   * @returns {Promise<EnumKey<typeof, databaseAdmin.spanner.admin.database.v1.Backup.State> | undefined>}
+   *     when resolved, contains the current state of the backup if it exists, or
+   *     undefined if the backup does not exist.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backup = instance.backup('my-backup');
+   * const state = await backup.getState();
+   * const backupCompleted = (state === 'READY');
+   */
+  async getState(): Promise<
+    | EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State>
+    | undefined
+  > {
+    try {
+      const [backupInfo] = await this.getMetadata();
+      return backupInfo.state || undefined;
+    } catch (err) {
+      if (err.code === status.NOT_FOUND) {
+        return undefined;
```

Comment:
Is this better than returning the 404? It would make more sense to me to return the error that the backup doesn't exist. Giving "undefined" or any other error could lead the user to believe the backup exists, but the state isn't readable yet. They could repeat the request thinking it was a system glitch or something temporary instead of having the correct diagnosis.

Reply:
Great idea. Done.